nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -371,9 +391,9 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
     private ValidationResult validateDynamicKCLConfigProperty(final String subject, final String input, final ValidationContext context) {
         final ValidationResult.Builder validationResult = new ValidationResult.Builder().subject(subject).input(input);
 
-        if (!subject.matches("^(?!with)[a-zA-Z]\\w*$")) {
+        if (!subject.matches("^(?!with)[a-zA-Z][a-zA-Z0-9_]*(\\.[a-zA-Z][a-zA-Z0-9_]*)?$")) {
```

Comment:
```suggestion
        if (!subject.matches("^(?!with)[a-zA-Z]\\w*(\\.[a-zA-Z]\\w*)?$")) {
```

Reply:
Is the !`with` prefix restriction still relevant with the updated Kinesis Client?

Reply:
No, but if someone is still trying to use the old methods, this could still help them catch the error.  None of the new methods will begin with "with" either.

Reply:
I guess that's a fair point, no object to the `(?!with)` prefix check staying in the RegEx